src/mod_antispam_rtbl.erl

```diff
@@ -114,33 +128,127 @@ parse_items(Items) ->
 retract_items(Ids) ->
     lists:foldl(fun(ID, Acc) -> Acc#{ID => false} end, #{}, Ids).
 
--spec service_jid(binary()) -> jid().
-service_jid(Host) ->
-    jid:make(<<>>, Host, <<?SERVICE_JID_PREFIX, (ejabberd_cluster:node_id())/binary>>).
-
 %%--------------------------------------------------------------------
-%% Hook callbacks.
+%%| Handle iq_reply
+
+%% TODO: How to know the host and node when timeout?
+%% handle_iq_reply(timeout, blocked_domains, State) ->
```

Comment:
This is important because if there‘s a dns lookup failure or the receiving host does not answer, the service will be broken.

Reply:
That said, there should be a test for that. Maybe you could add that?